plugins/test/unit/plugins/profilers/profiler_mocks.py

```diff
@@ -22,7 +22,10 @@ def get_units(repo_id, criteria=None):
             for u in existing_units:
                 if criteria:
                     if u.type_id in criteria.type_ids:
-                        ret_val.append(u)
+                        if u.unit_key == criteria.unit_filters:
```

Comment:
Whoa, intense.  :scream_cat:

I know this isn't your code, but could you add some comments for this?